### PR TITLE
(Fixes #150) The `whenMet` method returns DataNode instead of S3DataNode

### LINKED ISSUE #150: The `whenMet` method returns DataNode instead of S3DataNode

### PR DESCRIPTION
**Pull request by sethyates**

### CODE REVIEW

File: core/src/main/scala/com/krux/hyperion/datanode/S3DataNode.scala [hunk 1]
```diff
@@ -16,11 +16,16 @@ sealed trait S3DataNode extends Copyable {
   def asOutput(): String = asOutput(1)
   def asOutput(n: Integer): String = "${" + s"OUTPUT${n}_STAGING_DIR}"
 
-  def withDataFormat(fmt: DataFormat): S3DataNode
-
   def named(name: String): S3DataNode
   def groupedBy(client: String): S3DataNode
 
+  def withDataFormat(fmt: DataFormat): S3DataNode
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/datanode/S3DataNode.scala]: I'm not sure, but does it work if you change the `whenMet` in `DataNode` to
````scala
def whenMet(conditions: Precondition*): this.type
```

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/datanode/S3DataNode.scala]: I'll give it a go.

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/datanode/S3DataNode.scala]: Doesn't seem very joyous.  Can I defer that for later?

**Comment by realstraw** [comment]
:+1: :sparkles: :sparkles: :sparkles:

**Pull request merged**